### PR TITLE
[CHK-1565] feat(monitoring): correlate event dispatcher async processing with original endpoint

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

* correlate event dispatcher async processing with original endpoint

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/94
Depends on #107

### CODE REVIEW

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt [hunk 1]
```diff
@@ -474,6 +480,48 @@ fun <T> runPipelineWithDeadLetterQueue(
     }
 }
 
+fun <T> runTracedPipelineWithDeadLetterQueue(
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt:483]: Can we remove `runPipelineWithDeadLetterQueue` function?

**Review by pietro-tota — DISMISSED** [review]
Well done! Just a comment

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt (hunk 1 above)
Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt:483]: Not until all consumers integrate tracing propagation, ie PR #118

**Review by pietro-tota — DISMISSED** [review]
Well done, just remember to restore ecommerce common branch

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionExpirationQueueConsumer.kt [hunk 2]
```diff
@@ -47,24 +52,27 @@ class TransactionExpirationQueueConsumer(
...
   @ServiceActivator(inputChannel = "transactionexpiredchannel", outputChannel = "nullChannel")
   fun messageReceiver(
     @Payload payload: ByteArray,
-    @Header(AzureHeaders.CHECKPOINTER) checkPointer: Checkpointer
+    @Header(AzureHeaders.CHECKPOINTER) checkPointer: Checkpointer,
+    @Headers headers: MessageHeaders
   ): Mono<Void> {
     val binaryData = BinaryData.fromBytes(payload)
-    val transactionId = getTransactionIdFromPayload(binaryData)
+    val queueEvent =
```
[the remark this answers is not in the document]
Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionExpirationQueueConsumer.kt]: Fixed in [15b8cdc](https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/108/commits/15b8cdcb13b1b620db381e5d4672dec0264cbe4c)

File: pom.xml [hunk 3]
```diff
@@ -479,8 +508,8 @@
 				<configuration>
 					<skipCheckoutIfExists>false</skipCheckoutIfExists>
 					<connectionUrl>scm:git:https://github.com/pagopa/pagopa-ecommerce-commons.git</connectionUrl>
-					<scmVersionType>tag</scmVersionType>
-					<scmVersion>${pagopa-ecommerce-commons.version}</scmVersion>
+					<scmVersionType>branch</scmVersionType>
```
[the remark this answers is not in the document]
Comment (giovanniberti) [reply pom.xml]: Done in [a1906ab](https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/108/commits/a1906abf0e67efaae459e94358fb6b39bc8f2208)

**Pull request closed without merging**